python/pyspark/sql/catalog.pyi

```diff
@@ -53,7 +54,7 @@ class Catalog:
     def dropGlobalTempView(self, viewName: str) -> None: ...
     def registerFunction(
         self, name: str, f: Callable[..., Any], returnType: DataType = ...
-    ) -> None: ...
+    ) -> UserDefinedFunctionLike: ...
```

Comment:
Note for reviewer:
https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/python/pyspark/sql/catalog.py#L257

https://github.com/apache/spark/blob/ff1fc5ed4b685b4f5f83d5f600b24f089dd4522e/python/pyspark/sql/udf.pyi#L46-L51